lib/phobos/handler.rb

```diff
@@ -4,24 +4,24 @@ def self.included(base)
       base.extend(ClassMethods)
     end
 
-    def before_consume(payload)
+    def before_consume(payload, metadata)
```

Comment:
This is a breaking change. I would like to have access to the metadata inside of before_consume for logging purposes (right now i'm digging into the Phobos.config to get that information). My preference would be to instantiate the handler with the metadata so there's no need to pass it as an arg to `before_consume`, `around_consume`, or `consume`:

```
def process_message(payload)
  instrument('listener.process_message', @metadata) do
    handler = @listener.handler_class.new(@metadata)
    consume_block = Proc.new { handler.consume(preprocessed_payload) }

    if @listener.handler_class.respond_to?(:around_consume)
      # around_consume class method implementation
      @listener.handler_class.around_consume(preprocessed_payload, &consume_block)
    else
      # around_consume instance method implementation
      handler.around_consume(preprocessed_payload, &consume_block)
    end
  end
end
    ...
```
However, this is a larger breaking change. Open other perspectives on this.